os/fs/procfs/fs_procfsereport.c

```diff
@@ -0,0 +1,579 @@
+
+static int ereport_open(FAR struct file *filep, FAR const char *relpath, int oflags, mode_t mode)
+{
+	FAR struct ereport_file_s *efile;
+	FAR const struct ereport_node_s *node;
+
+	fvdbg("Open '%s'\n", relpath);
+
+	/* PROCFS is read-only.  Any attempt to open with any kind of write
+	 * access is not permitted.
+	 *
+	 * REVISIT:  Write-able proc files could be quite useful.
+	 */
+
+	if ((oflags & O_WRONLY) != 0 || (oflags & O_RDONLY) == 0) {
+		fdbg("ERROR: Only O_RDONLY supported\n");
+		return -EACCES;
+	}
+
+	/* Find the nodes of connectivity directory */
+	node = ereport_findnode(relpath);
+	if (!node) {
+		/* Entry not found */
+		fdbg("ERROR: Invalid path \"%s\"\n", relpath);
+		return -ENOENT;
+	}
+
+	if (!DIRENT_ISFILE(node->dtype)) {
+		fdbg("ERROR: Path \"%s\" is not a regular file\n", relpath);
+		return -EISDIR;
+	}
+
+	/* Allocate a container to hold the domain selection */
+	efile = (FAR struct ereport_file_s *)kmm_zalloc(sizeof(struct ereport_file_s));
+	if (!efile) {
+		fdbg("ERROR: Failed to allocate file container\n");
+		return -ENOMEM;
+	}
+
+	efile->node = node;
+
+	/* Save the index as the open-specific state in filep->f_priv */
+	filep->f_priv = (FAR void *)efile;
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: ereport_close
+ ****************************************************************************/
+
+static int ereport_close(FAR struct file *filep)
+{
+	FAR struct ereport_file_s *efile;
+
+	/* Recover our private data from the struct file instance */
+
+	efile = (FAR struct ereport_file_s *)filep->f_priv;
+	DEBUGASSERT(efile);
+
+	/* Release the file container structure */
+
+	kmm_free(efile);
+	filep->f_priv = NULL;
+	return OK;
+}
+
+/****************************************************************************
+ * Name: ereport_read
+ ****************************************************************************/
+
+static ssize_t ereport_read(FAR struct file *filep, FAR char *buffer, size_t buflen)
+{
+	FAR struct ereport_file_s *efile;
+	ssize_t ret;
+
+	fvdbg("buffer=%p buflen=%d\n", buffer, (int)buflen);
+
+	/* Recover our private data from the struct file instance */
+
+	efile = (FAR struct ereport_file_s *)filep->f_priv;
+	DEBUGASSERT(efile);
+
+	/* Provide the requested data */
+	ret = 0;
+
+	switch (efile->node->nodetype) {
+	case EREPORT_TASKADDR:			/* Task Address */
+		ret = ereport_taskaddr_read(efile, buffer, buflen, filep->f_pos);
+		break;
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	/* Update the file offset */
+
+	if (ret > 0) {
+		filep->f_pos += ret;
+	}
+
+	return ret;
+}
+
+/****************************************************************************
+ * Name: ereport_dup
+ *
+ * Description:
+ *   Duplicate open file data in the new file structure.
+ *
+ ****************************************************************************/
+static int ereport_dup(FAR const struct file *oldp, FAR struct file *newp)
```

Comment:
You defined dup as a null, aren't you?

Reply:
Yes, indeed. Will remove the implementation in this comment. Thanks for spotting.